pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1780,6 +1780,31 @@ private void updateCurrentDocumentCountMetrics() {
     }
   }
 
+  /**
+   * Creates a {@link StreamMessageDecoder} using properties in {@link StreamConfig}.
+   *
+   * @param streamConfig The stream config from the table config
+   * @param fieldsToRead The fields to read from the source stream
+   * @return The initialized StreamMessageDecoder
+   */
+  private StreamMessageDecoder create(Set<String> fieldsToRead) {
+    String decoderClass = _streamConfig.getDecoderClass();
+    try {
+      Map<String, String> decoderProperties = _streamConfig.getDecoderProperties();
+    StreamMessageDecoder decoder = PluginManager.get().createInstance(decoderClass);
+    try {
+      decoder.init(decoderProperties, fieldsToRead, _streamConfig.getTopicName());
+    } catch (UnsupportedOperationException e) {
+      // Backward compatibility
+      decoder.init(fieldsToRead, _streamConfig, _tableConfig, _schema);
+    }
```

Comment:
With the default fallback in the interface, we should just invoke the new method